File: budget-backend/src/main/java/com/example/budget/model/Category.java
package com.example.budget.model;

import com.example.budget.model.enumeration.CategoryType;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name="categories")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private CategoryType type;

    @ManyToOne(cascade = CascadeType.PERSIST)
    private User user;
}

File: budget-frontend/src/app/shared/types/category.ts
import {IUser} from './user';

export interface ICategory {
  id: number;
  name: string;
  type: string[];
  userDto: IUser;
}
// export class Category implements ICategory {
//   id: number;
//   name: string;
//   type: string[];
//   userDto: User;
// }
